actions/signed-commits/src/git/github-git/repo-tags.ts

```diff
@@ -206,21 +267,23 @@ export function getMajorVersionTags(
  * @returns The parsed tag
  */
 export function parseTagName(tagName: string, separator: string) {
-  // [a-z-]+ is the package name
-  // (\d+) is the major/minor/patch version
-  const tagRegex = new RegExp(`([a-z0-9-]+)${separator}(\\d+).(\\d+).(\\d+)`);
+  // [a-z0-9-]+ is the package name
+  // (\\d+) is the major/minor/patch version
+  const escapedSeparator = separator.replace(/[.*+?^${}()|[\]\\]/g, "\\\\$&");
+  const tagRegex = new RegExp(
+    `^([a-z0-9-]+)${escapedSeparator}(\\d+)\\.(\\d+)\\.(\\d+)`,
+  );
   const match = tagRegex.exec(tagName);
   if (!match || match.length < 5) {
+    core.debug(
+      `parseTagName: No match for tagName "${tagName}" with separator "${separator}" using regex "${tagRegex}"`,
+    );
     return;
   }
 
   const name = match[1];
   const version = `${match[2]}.${match[3]}.${match[4]}`;
   const majorVersion = match[2] ?? "0";
-  if (majorVersion == "0") {
```

Comment:
@erikburt I moved this out into https://github.com/smartcontractkit/.github/pull/1059/files#diff-a4c32260891872474d347dd3de486467d89e31d27876c05e2dd2877986d545b5R238